### PR TITLE
Switch to 3box comments

### PR DESCRIPTION
**Pull request by tibetsprague**
Fixes https://github.com/daostack/alchemy/issues/1553

- removes App.spec.ts because it was [failing due to this error](https://github.com/ipfs-shipyard/ipfs-companion/issues/475).  It contained no tests anyway.  And merely attempting to import a react component in the absence of a browser seems like a weird thing to do.

### CODE REVIEW

**Comment by EzraWeller** [comment]
This is working pretty well for me right now (https://alchemy-staging-kovan.herokuapp.com/). Though it's actually on rinkeby, not kovan?

The styling of the warning ("We are moving from Disqus to 3Box...") is a bit strange, but I'm assuming we'll remove that in a couple weeks anyway.

**Comment by orenyodfat** [comment]
please use latest 3Box v1.19.0

**Comment by dkent600** [comment]
I created this ticket on 3Box regarding the crash described above:  https://github.com/3box/3box-comments-react/issues/34

**Comment by dkent600** [comment]
The fix from 3box is in:  https://github.com/3box/3box-comments-react/issues/34

**Comment by dkent600** [comment]
I have improved the warning message text copy and appearance

**Comment by dkent600** [comment]
When I first come into Genesis Alpha, as Andrew Smith (0xffcf8fdee72ac11b5c542428b35eef5769c409f0) I see this "comments" spinny button spinning forever.

image

**Comment by dkent600** [comment]
The message should be showing the account address if we know the user's profile:

image

image

**Comment by dkent600** [comment]
What is this hard-coded address?  Should be in the heroku configuration?

```
          <ThreeBoxComments
            spaceName="DAOstack"
            threadName={daoState.id}
            adminEthAddr={"0x0084FB1d84F2359Cafd00f92B901C121521d6809"}
```

**Comment by dkent600** [comment]
This 3Box promotional text is not properly aligned horizontally.  Should be centered.

image

**Comment by orenyodfat** [comment]
> What is this hard-coded address? Should probably be in the heroku configuration.
> 
> ```
>           <ThreeBoxComments
>             spaceName="DAOstack"
>             threadName={daoState.id}
>             adminEthAddr={"0x0084FB1d84F2359Cafd00f92B901C121521d6809"}
> ```
> 
> @orenyodfat @EzraWeller Do you recall whose address this is?
> 
It is the 3box thread moderator address.
I suggest it should be @EzraWeller address and have a 3box profile behind it .
Also , I hope the space name is not already been taken by 0x0084FB1d84F2359Cafd00f92B901C121521d6809 (which I am not sure who is it )
read https://medium.com/3box/introducing-3box-comments-plugin-c6a882d11f4a .

**Comment by dkent600** [comment]
> > What is this hard-coded address? Should probably be in the heroku configuration.
> > ```
> >           <ThreeBoxComments
> >             spaceName="DAOstack"
> >             threadName={daoState.id}
> >             adminEthAddr={"0x0084FB1d84F2359Cafd00f92B901C121521d6809"}
> > ```
> > 
> > 
> > @orenyodfat @EzraWeller Do you recall whose address this is?
> 
> It is the 3box thread moderator address.
> I suggest it should be @EzraWeller address and have a 3box profile behind it .
> Also , I hope the space name is not already been taken by 0x0084FB1d84F2359Cafd00f92B901C121521d6809 . [Tibet](https://alchemy.daostack.io/profile/0x0084fb1d84f2359cafd00f92b901c121521d6809?daoAvatarAddress=0x294f999356ed03347c7a23bcbcf8d33fa41dc830) .
> read https://medium.com/3box/introducing-3box-comments-plugin-c6a882d11f4a .

Yeah, thanks I knew all that already, was just wondering who that address belongs to.  And yeah, who it should be.  Note that I have made it into an env variable in Heroku.

**Comment by orenyodfat** [comment]
> > > What is this hard-coded address? Should probably be in the heroku configuration.
> > > ```
> > >           <ThreeBoxComments
> > >             spaceName="DAOstack"
> > >             threadName={daoState.id}
> > >             adminEthAddr={"0x0084FB1d84F2359Cafd00f92B901C121521d6809"}
> > > ```
> > > 
> > > 
> > > @orenyodfat @EzraWeller Do you recall whose address this is?
> > 
> > 
> > It is the 3box thread moderator address.
> > I suggest it should be @EzraWeller address and have a 3box profile behind it .
> > Also , I hope the space name is not already been taken by 0x0084FB1d84F2359Cafd00f92B901C121521d6809 . [Tibet](https://alchemy.daostack.io/profile/0x0084fb1d84f2359cafd00f92b901c121521d6809?daoAvatarAddress=0x294f999356ed03347c7a23bcbcf8d33fa41dc830) .
> > read https://medium.com/3box/introducing-3box-comments-plugin-c6a882d11f4a .
> 
> Yeah, thanks I knew all that already, was just wondering who that address belongs to. And yeah, who it should be. Note that I have made it into an env variable in Heroku.

just wonder , why env variable  and not a constant in json ? it is a one time set .

**Comment by dkent600** [comment]
Sometimes it takes 15-20 seconds on rinkeby for 3Box to fetch the comments for display.

**Comment by orenyodfat** [comment]
> Sometimes it takes 15-20 seconds on rinkeby for 3Box to fetch the comments for display.

this is not reasonable .  can you ask 3box re that ? @dkent600

File: src/settings.ts [hunk 1]
```diff
@@ -166,6 +166,7 @@ function getWeb3ConnectProviderOptions(network: string) {
 }
 
 export const settings = {
+  "3BoxCommentsAdmin": "0x0084FB1d84F2359Cafd00f92B901C121521d6809",
```
Comment (orenyodfat) [inline src/settings.ts:169]: this suppose to be @EzraWeller address

**Comment by dkent600** [comment]
1. go to Alchemy with a cached but not connected account and with metamask pointing to a network != rinkeby
1. go to a DAO landing page
1. try to enter a comment

RESULT:
* Fails to connect to metamask because of wrong network
* 3box is hung with a spinning icon

EXPECT:
The attempt to submit the comment should be aborted.

**Comment by dkent600** [comment]
When entering a comment and my account is already connected properly to the network, 3Box asks for my signature and then hangs.

**Comment by dkent600** [comment]
STEPS:
1. go to Alchemy with a cached but _not connected_ account and with metamask pointing properly to rinkeby
1. go to a DAO landing page
1. try to enter a comment

RESULT:
* Fails to connect to metamask because of wrong network
* 3box is hung with a spinning icon

EXPECT:
The attempt to submit the comment should be aborted.

**Comment by dkent600** [comment]
When there is no cached account, 3Box should not show its own login button.  We don't need yet another way to support logging in.

image

**Comment by dkent600** [comment]
Note:  apparently 3box-comments-react@3.0.4 does not work with 3box@1.20.0

**Comment by dkent600** [comment]
3BOX label to the right of the edit box, that show when the spinner is there, is too far to the right and also causes a horizontal scrollbar to appear

image

**Comment by dkent600** [comment]
3Box has some absolutely horrible CSS in the UI it injects into Alchemy.  I'm having to hack some CSS overrides.  (classic issues with use of CSS absolute positioning.)

**Comment by dkent600** [comment]
3box is hanging in the call to `box.auth`.  At best it occasionally returns, but after minutes.  This could account for the hangs I reported in other comments above.  I have reported it to 3box in a ticket here:  https://github.com/3box/3box-js/issues/848

**Comment by dkent600** [comment]
I have been informed by 3box that 3box-comments-react is only barely being supported at the moment.

**Comment by dkent600** [comment]
Upon an account's first attempt to enter a comment, the entire process requires four prompts for a signature, half of which seem to be redundant, and a couple fairly long pauses in rinkeby which I expect will be even longer on mainnet.

**Comment by dkent600** [comment]
STEPS:

1. create a comment in account 1

**Comment by dkent600** [comment]
Closing, we will start from scratch using 3Box Threads

**Pull request closed without merging**